shared/src/business/entities/cases/Case.js

```diff
@@ -1018,7 +1018,15 @@ Case.prototype.addDocketEntry = function (docketEntryEntity) {
 
     if (updateIndex) {
       docketEntryEntity.index = this.generateNextDocketRecordIndex();
+    } else if (
```

Comment:
💭 I wonder if it would be more testable and reliable to put code like this into the Joi validation for DocketEntry? Using a Joi conditional we could specify that when `eventCode` is `STIN`, `isOnDocketRecord` has to be false

Just a thought, let me know what you think!

Reply:
Oooh same with the index for `STIN` has to be 0? That way if we ever try to persist that value the validation will throw an error. Again, just a thought.

Reply:
I thought the intent of using TypeScript was to move away from Joi validations. Happy to add it if you think it's important though.